Functions/Administration/xASL_adm_GetPhilipsScaling.m

```diff
@@ -29,45 +30,59 @@
 rescaleSlopeNifti = header.dat.scl_slope;
 
 if isfield(parms,'RWVSlope')
-	% If the RealWorldValue is present, then dcm2nii scales to them and ignores everything else
-    fprintf('=========================================== Scaling ==========================================\n');
-	if length(parms.RWVSlope)>1
-		[~,idx] = min(abs(parms.RWVSlope - rescaleSlopeNifti));
-		parms.RWVSlope = parms.RWVSlope(idx);
-		if isfield(parms,'RescaleSlopeOriginal') && length(parms.RescaleSlopeOriginal)>1
-			parms.RescaleSlopeOriginal = parms.RescaleSlopeOriginal(idx);
+	% Obtains dcm2niix version used for conversion and ExploreASL version
+	% If ExploreASL version >= 1.10.0 or dcm2niix version >= 20210317
+	if isfield(parms,'ConversionSoftware') && (strcmp(parms.ConversionSoftware,'dcm2nii') || strcmp(parms.ConversionSoftware,'dcm2niix')) && isfield(parms,'ConversionSoftwareVersion')
+		dcm2niixVersion = parms.ConversionSoftwareVersion;
+		dcm2niixVersion = str2num(dcm2niixVersion(end-7:end));
+	else
+		dcm2niixVersion = 0;
+	end
+	
+	% It has to use the correct dcm2niix version that is able to handle this parameter and it also needs to report that it did do that
+	if dcm2niixVersion >= 20210317 && isfield(parms,'UsePhilipsFloatNotDisplayScaling') && (parms.UsePhilipsFloatNotDisplayScaling == 1)
+		scaleFactor = 0;
+	else
+		% If the RealWorldValue is present, then dcm2nii scales to them and ignores everything else
+		fprintf('=========================================== Scaling ==========================================\n');
+		if length(parms.RWVSlope)>1
+			[~,idx] = min(abs(parms.RWVSlope - rescaleSlopeNifti));
+			parms.RWVSlope = parms.RWVSlope(idx);
+			if isfield(parms,'RescaleSlopeOriginal') && length(parms.RescaleSlopeOriginal)>1
+				parms.RescaleSlopeOriginal = parms.RescaleSlopeOriginal(idx);
+			end
+			
+			if isfield(parms,'RescaleSlope') && length(parms.RescaleSlope)>1
+				parms.RescaleSlope = parms.RescaleSlope(idx);
+			end
+			
+			if isfield(parms,'MRScaleSlope') && length(parms.MRScaleSlope)>1
+				parms.MRScaleSlope = parms.MRScaleSlope(idx);
+			end
 		end
 		
-		if isfield(parms,'RescaleSlope') && length(parms.RescaleSlope)>1
-			parms.RescaleSlope = parms.RescaleSlope(idx);
+		if ~isempty(parms.RWVSlope)
+			if ~isnear(parms.RWVSlope,rescaleSlopeNifti,parms.RWVSlope/100) && (rescaleSlopeNifti ~= 1)
```

Comment:
Here you only print stuff, you don't actually execute what you print here that you will do.

I would put the execution in the same if-statement as the printing, so we are sure that the messages on the screen/log file represent what was actually done.

Reply:
We are sure - the execution is 5 lines below as `RWVSlope` is always used in this paragraph. So no point writing it five times in each branch, when it is executed in all branches.

Reply:
OK then only improve the comment on line 92 "set the scaling to remove"

Reply:
Done 337e00c4